File: design_patterns/python/adapter/adapter.py
class Adapter:

    def __init__(self):
        self._adaptee = Adaptee()

    def request(self):
        self._adaptee.specific_request()


class Adaptee:

    def specific_request(self):
        print("Specific method is used.")


if __name__ == "__main__":
    adapter = Adapter()
    adapter.request()

File: design_patterns/csharp/Adapter/Domain/Adapter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Adapter.Domain
{
    class Adapter : ITarget
    {
        Adaptee _adaptee;
        public Adapter()
        {
            this._adaptee = new Adaptee();
        }


        public void Request()
        {
            this._adaptee.SpecificRequest();
        }
    }
}
